lib/logstash/inputs/kafka.rb

```diff
@@ -374,6 +374,8 @@ def maybe_set_metadata(event, record)
     end
     if @metadata_mode.include?(:headers)
       record.headers.each do |header|
+        next if header.nil? || header.value.nil? || header.key.nil?
```

Comment:
I tried to comment this line and ran the test, but was till green. Seems the test doesn't enter the loop.

Reply:
Great that you spot it